integration-tests/test_client_options.py

```diff
@@ -17,3 +21,22 @@ def test_set_ansible_host_info(insights_client):
     ret = insights_client.run("--ansible-host=foo.example.com", check=False)
     assert "Could not update Ansible hostname" not in ret.stdout
     assert ret.returncode == 0
+
+
+def test_no_upload(insights_client):
+    """Verify no traceback is returned when running '--no-upload' command
+    and the archive is created and saved on the system
+    """
+    archive_saved = "Archive saved at"
+    upload_message = "Successfully uploaded report"
+
+    insights_client.register()
+
+    archive_file_before = glob.glob(ARCHIVE_CACHE_DIRECTORY)
```

Comment:
globbing a directory will always return 1, as the directory itself matches the glob exactly :D

```suggestion
    archive_file_before = glob.glob(f"{ARCHIVE_CACHE_DIRECTORY}/*.tar.gz")
```

Reply:
done